### PR TITLE
Fix Date parsing crash when parsing a negative year

### LINKED ISSUE #50328: error when using `DateTime` with string input when year is negative
So I tried to use `DateTime("-2011-01-01", "yyyy-mm-dd")` and ran into this error:
```julia
julia> DateTime("-2011-01-01", "yyyy-mm-dd")
ERROR: ArgumentError: Unable to parse date time. Expected directive DatePart(yyyy) at char 1
Stacktrace:
 [1] macro expansion
   @ ~/.julia/juliaup/julia-1.9.1+0.x64.linux.gnu/share/julia/stdlib/v1.9/Dates/src/parse.jl:104 [inlined]
 [2] tryparsenext_core(str::String, pos::Int64, len::Int64, df::DateFormat{Symbol("yyyy-mm-dd"), Tuple{Dates.DatePart{'y'}, Dates.Delim{Char, 1}, Dates.DatePart{'m'}, Dates.Delim{Char, 1}, Dates.DatePart{'d'}}}, raise::Bool)
   @ Dates ~/.julia/juliaup/julia-1.9.1+0.x64.linux.gnu/share/julia/stdlib/v1.9/Dates/src/parse.jl:38
 [3] macro expansion
   @ ~/.julia/juliaup/julia-1.9.1+0.x64.linux.gnu/share/julia/stdlib/v1.9/Dates/src/parse.jl:150 [inlined]
 [4] tryparsenext_internal
   @ ~/.julia/juliaup/julia-1.9.1+0.x64.linux.gnu/share/julia/stdlib/v1.9/Dates/src/parse.jl:125 [inlined]
 [5] parse(::Type{DateTime}, str::String, df::DateFormat{Symbol("yyyy-mm-dd"), Tuple{Dates.DatePart{'y'}, Dates.Delim{Char, 1}, Dates.DatePart{'m'}, Dates.Delim{Char, 1}, Dates.DatePart{'d'}}})
   @ Dates ~/.julia/juliaup/julia-1.9.1+0.x64.linux.gnu/share/julia/stdlib/v1.9/Dates/src/parse.jl:284
 [6] DateTime(dt::String, format::String; locale::Dates.DateLocale)
   @ Dates ~/.julia/juliaup/julia-1.9.1+0.x64.linux.gnu/share/julia/stdlib/v1.9/Dates/src/io.jl:555
 [7] DateTime(dt::String, format::String)
   @ Dates ~/.julia/juliaup/julia-1.9.1+0.x64.linux.gnu/share/julia/stdlib/v1.9/Dates/src/io.jl:554
 [8] top-level scope
   @ REPL[6]:1
```

The functions works fine if the year isn't negative. I know what I'm trying to do can be achieved by doing this instead:
```julia
julia> DateTime(-2011, 01, 01, 0)
-2011-01-01T00:00:00
```
But shouldn't the former way of doing things work as well?

Julia Verion 1.9.1 installed using juliaup:
```julia
julia> versioninfo()
Julia Version 1.9.1
Commit 147bdf428cd (2023-06-07 08:27 UTC)
Platform Info:
  OS: Linux (x86_64-linux-gnu)
  CPU: 12 × AMD Ryzen 5 4600H with Radeon Graphics
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-14.0.6 (ORCJIT, znver2)
  Threads: 1 on 12 virtual cores
```

### PR DESCRIPTION
**Pull request by Sr0liveira02**
Fixes #50328.
Originally checked on version 1.10.0 but still relevant in the current version in master

Bug: In method Date(str), when given a negative year as an argument (ex: "-2000"), it would output an error while in Date(int) once given a negative year (ex: "-2000") it would work as intended returning "-2000-01-01".

Fix: In function tryparsenext_base10 the character '-' wasn't being accounted so i check if it existed in the first iteration of the string and if yes I would multiply the output * -1.

Test: Added some tests that verify this specific Fix.

### CODE REVIEW

File: stdlib/Dates/test/io.jl [hunk 1]
```diff
@@ -618,4 +618,16 @@ end
...
+@testset "Issue #50328: parsing negative years" begin
+    @test Date("-2013-10-10") == Date(-2013, 10, 10)
+    @test Date("-2013") == Date(-2013, 01, 01)
+
+    try
+        Date("")
+        @test false
+    catch err
+        @test err.msg == "Cannot parse an empty string as a Date or Time"
+    end
```
Comment (quinnj) [inline stdlib/Dates/test/io.jl:630]: unnecessary?

Comment (vtjnash) [inline stdlib/Dates/test/io.jl:630]:
```suggestion
    @test_throws "Cannot parse an empty string as a Date or Time" Date("")
```

**Comment by ViralBShah** [comment]
Bump.

**Comment by Sr0liveira02** [comment]
Hi, sorry for the delay.
I understand that the test might have been unnecessary so I removed it.

**Comment by ViralBShah** [comment]
@quinnj Good to merge?

**Review by quinnj — APPROVED** [review]

**Pull request merged**